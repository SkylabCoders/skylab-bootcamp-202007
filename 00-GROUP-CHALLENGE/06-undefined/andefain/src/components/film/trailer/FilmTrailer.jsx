import React from 'react';
import { Link } from 'react-router-dom';
import './trailer.scss';
import PropTypes from 'prop-types';

function FilmTrailer({ trailer }) {
	debugger;
	return (
		<div className="trailer-container">
			<Link to="/film">
				<img alt="Poster" src={trailer.image} className="poster-img" />
			</Link>
			<Link to="/film">
				<img
					alt="Trailer"
					src={trailer.trailer ? trailer.trailer.thumbnailUrl : ''}
					className="trailer-img"
				/>
			</Link>
		</div>
	);
}

FilmTrailer.propTypes = {
	trailer: PropTypes.object
};

export default FilmTrailer;
