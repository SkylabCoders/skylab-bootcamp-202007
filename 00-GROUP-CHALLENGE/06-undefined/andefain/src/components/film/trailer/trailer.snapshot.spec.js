import renderer from 'react-test-renderer';
import React from 'react';
import FilmTrailer from './trailer';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Film trailer snapshot', () => {
	const props = {
		image:
			'https://imdb-api.com/images/original/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_Ratio0.6791_AL_.jpg',
		trailer: {
			thumbnailUrl:
				'https://m.media-amazon.com/images/M/MV5BNTdlZGFmMDQtM2RiYi00NzUxLThhZWQtN2M4ZmYwMzllZGYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_.jpg'
		}
	};

	const trailerTree = renderer.create(
		<Router>
			<FilmTrailer trailer={props} />
		</Router>
	);
	it('Should match', () => {
		expect(trailerTree.toJSON()).toMatchSnapshot();
	});
});
