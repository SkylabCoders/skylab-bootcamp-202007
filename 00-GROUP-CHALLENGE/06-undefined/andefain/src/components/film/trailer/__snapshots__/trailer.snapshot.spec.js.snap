// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film trailer snapshot Should match 1`] = `
<div
  className="trailer-container"
>
  <a
    href="/film"
    onClick={[Function]}
  >
    <img
      alt="Poster"
      className="poster-img"
      src="https://imdb-api.com/images/original/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_Ratio0.6791_AL_.jpg"
    />
  </a>
  <a
    href="/film"
    onClick={[Function]}
  >
    <img
      alt="Trailer"
      className="trailer-img"
      src="https://m.media-amazon.com/images/M/MV5BNTdlZGFmMDQtM2RiYi00NzUxLThhZWQtN2M4ZmYwMzllZGYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_.jpg"
    />
  </a>
</div>
`;
